src/ray/stats/metric.h

```diff
@@ -29,30 +29,42 @@ namespace stats {
 /// Include tag_defs.h to define tag items
 #include "ray/stats/tag_defs.h"
 
+/// StatsConfig per process.
+/// Note that this is not thread-safe. Don't modify its internal values
+/// outside stats::Init() or stats::Shutdown() method.
```

Comment:
Initial reason was because of deadlock (since `stats::Init` and `StatsConfig` can cause deadlocks). This can be solved if we combine all of them within a single class or remove all statsConfig setter. Maybe I can try the latter approach for now. (because it takes less time)